ADR/nnnn-task-versioning.md

```diff
@@ -0,0 +1,361 @@
+automatic. Appendix 2: [In defense of manual versioning](#in-defense-of-manual-versioning)
+argues that automatic versioning is not a good idea.
+
+## Appendix
+
+### The technicals of Task versioning
+
+#### Repository layout
+
+The current versioning practices for Konflux Tasks require a repository layout
+with separate directories for each `major.minor` version:
+
+```text
+task
+└── hello
+    ├── 0.1
+    │   └── hello.yaml
+    ├── 0.2
+    │   └── hello.yaml
+    ├── 1.0
+    │   └── hello.yaml
+    └── 1.1
+        └── hello.yaml
+```
+
+Updating a Task's version involves copy-pasting the Task definition and related resources
+into a new directory. This brings problems:
+
+- Reviewing changes is much harder than it should be. The Pull Request doesn't show
+  a diff, it shows a whole new file. The PR author can work around this by structuring
+  their commits intelligently, but it's a point of friction.
+- It's too easy to introduce "regressions" by merging features/bugfixes into the
+  wrong Task version. You open a PR for `task/hello/1.1`. In the meantime, someone
+  else creates `task/hello/1.2`. Both PRs merge with no conflicts, because as far
+  as git is concerned, the files have nothing to do with each other. Version 1.2
+  is now missing a bugfix from version 1.1.
+- It's enough of a chore to discourage developers from versioning.
+
+The suggested repository layout going forward would be:
+
+```text
+task
+└── hello
+    └── hello.yaml    (version defined only by the app.kubernetes.io/version label)
+```
+
+In case you want to maintain more than one "version stream" at a time:
+
+```text
+task
+└── hello
+    ├── 1.x
+    │   └── hello.yaml  <- branched off into separate dir when moving to 2.0
+    └── hello.yaml      <- latest 2.x
+```
+
+#### CI setup
+
+We want to make sure each meaningful change to a Task definition results in a version
+update. We may want to leave maintainers/contributors some freedom to skip version
+updates for changes they don't consider meaningful (e.g. formatting, description
+texts).
+
+A flexible way to achieve that would be to **release Tasks only when their version
+number changes** (don't even build the Task bundle otherwise).
+
+The CI should prompt the contributor to consider updating the version and the CHANGELOG.
+It should explain that the Task will not get released unless the version changes.
+
+> 💡
+> AI code review might work well here. You would want to teach the reviewer to decide
+> whether the change is meaningful and post a comment accordingly.
+
+#### Dealing with MintMaker updates
+
+Automated PRs for dependency updates may become a problem. They're already seen
+as a chore today by many Task maintainers. If manual action is required before
+a MintMaker PR can merge, the most likely outcome is that MintMaker PRs will stop
+getting merged.
+
+There's a few options to deal with this:
+
+- Do not consider them meaningful. Merge the PR without updating the changelog or
```

Comment:
> Do not consider them meaningful

Unless they break stuff. But then a developer need to step in and fix the problem updating the change log. So, it should work if we don't release changes without version bump.